Sources/Extensions/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -135,6 +135,45 @@ public extension Sequence {
         })
     }
 
+    /// SwifterSwift: Get the only element.
+    ///
+    ///     [].single() -> nil
+    ///     [4].single() -> 4
+    ///     [2, 4].single() -> nil
+    ///
+    /// - Returns: The only element in the array. If there are more elements, nil is returned. (optional)
+    public func single() -> Element? {
+        var singleElement: Element?
```

Comment:
Hey @andlang :))
A simpler implementation will be:
```swift 
public func single() -> Element? {
  guard count <= 1 else { return nil } 
  return self.first
}
```
I think is the same behavior but, I'm not sure, can you check if it is the same?

Reply:
IMHO it would just be better to not support this version of the extension. Really curious why we can't get the default closure to work

Reply:
In general the `default` closure works. The problem with it and `rethrows` is, that the compiler says you need to insert a `try` in your function call (`try ….single()`), since the condition is possibly throwable (even though the default `{ _ in true }` doesn’t throw).
But in having to add a `try`, the extension is difficult to use in the default cases:
```swift
if let result = (try? queryResult.single()) ?? nil {
   print(result)
}
```
vs.
```swift
if let result = queryResult.single() {
   print(result)
}
```

Reply:
@SD10 if you call it explicitly
```Swift
[].single({ _ in true})
```
you don't have to, but if you make it the default value of the param then you're forced to by the compiler. I'm also stumped as to why that's the case...

Reply:
@guykogus Can the closure be prefixed with `try`?

Reply:
No, compiler says that the closure is non-throwing. Seems that it's [a known bug in Swift](https://bugs.swift.org/browse/SR-1534).

Interestingly, it compiles if you make the closure optional:
```Swift
func single(where condition: ((Element) throws -> Bool)? = nil) rethrows -> Element? {
    var singleElement: Element?
    for element in self where (try condition?(element)) ?? true {
        guard singleElement == nil else {
            singleElement = nil
            break
        }
        singleElement = element
    }
    return singleElement
}
```
but then it crashes in runtime (also [a known bug in Swift](https://bugs.swift.org/browse/SR-313), it seems).

Reply:
@guykogus Thanks so much for doing that digging! 💯 I think my solution would be to just add this single version of the method without a default implementation until the compiler is fixed

Reply:
@BennX @guykogus Could you guys join the SwifterSwift Slack channel? I'm looking to upgrade the merge permissions of a few SwifterSwift members who have demonstrated a keen eye for detail over the past couple months. I just started a new job so I have a little less time than usual